### PR TITLE
feature(website): campaign page

### PR DESCRIPTION
**Pull request by andrashee**
Towards a campaign page. Still needs fine-tuning, but the payment association already works.

No contributions yet:
image

Some contributions:
image

No goals:
image

Last day to contribute:
image

1 day to contribute
image

Todos
- [ ] styling
- [ ] wording
- [ ] one-time contribution: pre-selected amounts
- [ ] monthly payments?

### CODE REVIEW

**Comment by ssandino** [comment]
Fabulous! Happy to help with text and styling inputs!

**Review by ssandino — APPROVED** [review]

File: website/src/app/[lang]/[region]/donate/campaign/[campaign]/page.tsx [hunk 1]
```diff
@@ -31,32 +34,85 @@ export default async function Page({ params }: CampaignPageProps) {
...
+							count: contributions,
+							amount: amountCollected,
+							currency: campaign.goal_currency,
+							total: campaign.goal,
+						},
+					})}
+				</Typography>
 			</div>
+			{daysLeft >= 0 && (
+				<Fragment>
```
Comment (mkue) [inline website/src/app/[lang]/[region]/donate/campaign/[campaign]/page.tsx:85]: nit: I always use `<></>`. But it's exactly the same...

File: website/src/app/[lang]/[region]/donate/campaign/[campaign]/page.tsx [hunk 2]
```diff
@@ -31,32 +34,85 @@ export default async function Page({ params }: CampaignPageProps) {
...
+					<Typography size="xl" weight="bold" color="accent">
+						{translator?.t('campaign.ended', { context: { count: daysLeft } })}
+					</Typography>
+				</div>
+			)}
 		</BaseContainer>
 	);
 }
+
+const daysUntilTs = (ts: Date): number => {
```
Comment (mkue) [inline website/src/app/[lang]/[region]/donate/campaign/[campaign]/page.tsx:115]: I would move this to `/shared/src/utils/date.ts`. That's where other date utility functions are located.

**Review by mkue — APPROVED** [review]
Very cool stuff @andrashee!

If we want to have this look nicer, I think we should first draw some mockups. For the moment, it looks good to me🙌🏼

**Comment by ssandino** [comment]
> Very cool stuff @andrashee!
> 
> If we want to have this look nicer, I think we should first draw some mockups. For the moment, it looks good to me🙌🏼

Agree! Very cool. @mkue @andrashee  I made some mockups in our [Figma file](https://www.figma.com/file/qGO3YI21AWIjWEyMPGUczM/Social-Income-Main-(Web%2C-App)?type=design&node-id=1870%3A5814&mode=design&t=sQ6uWuwjTEbgtC8B-1) (see page Campaign Page wip). For the moment just ideas about how to place things and what to include – but more as a starter to discuss, feel free to comment here or directly in the Figma file if easier. 

Here my favourite so far:
Generic Campaign Page

**Comment by ssandino** [comment]
@mkue @andrashee I asked @johanneswicki for input on the mockup. He has come up with another version ("ich habe hier versucht, das ganze etwas grosszügiger zu machen und damit mehr die künftige landing page zu matchen")

> If we want to have this look nicer, I think we should first draw some mockups.

Any inputs for a next version? 

Screenshot 2024-02-06 at 17 19 54

**Comment by andrashee** [comment]
Thanks a lot @ssandino, the mock-up look great! If you have time tomorrow, we could discuss some details. Would be free in the afternoon.

**Pull request merged**